From b0248185e2fdf9dc678e99aca1771892c456ffdd Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:15 -0400
Subject: [infiniband] ocrdma: report asic-id in query device

Message-id: <5bc979aa525ce6923772dd6318056149b4054c09.1411571481.git.dledford@redhat.com>
Patchwork-id: 96631
O-Subject: [Patch RHEL7 40/48] RDMA/ocrdma: report asic-id in query device
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitesh Ahuja <mitesh.ahuja@emulex.com>

Ocrdma does not report hw_ver when query_device is issued.  This patch
adds a meaningful value to this field.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Mitesh Ahuja <mitesh.ahuja@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 96c51abecc4379273fb751ff44eb57e4b243b5f0)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 90152de..acb434d 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -73,7 +73,7 @@ int ocrdma_query_device(struct ib_device *ibdev, struct ib_device_attr *attr)
  attr->page_size_cap = 0xffff000;
  attr->vendor_id = dev->nic_info.pdev->vendor;
  attr->vendor_part_id = dev->nic_info.pdev->device;
- attr->hw_ver = 0;
+ attr->hw_ver = dev->asic_id;
  attr->max_qp = dev->attr.max_qp;
  attr->max_ah = OCRDMA_MAX_AH;
  attr->max_qp_wr = dev->attr.max_wqe;
-- 
1.7.1